lib/types/string.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Soroban.Types.String do
+  @moduledoc """
+  `String` struct definition.
+  """
+
+  @behaviour Soroban.Types.Spec
+
+  alias Stellar.TxBuild.SCVal
+
+  defstruct [:value]
+
+  @type value :: binary()
+  @type t :: %__MODULE__{value: value()}
+
+  @impl true
+  def new(value) when is_binary(value) do
+    %__MODULE__{value: value}
+  end
```

Comment:
@Odraxs please use `, do:` notation in the `new/1` functions of the modules where possible, ex: Symbol, String